helix-term/src/commands.rs

```diff
@@ -2198,7 +2153,7 @@ pub fn join_selections(cx: &mut Context) {
         for line in lines {
             let mut start = text.line_to_char(line + 1).saturating_sub(1);
             let mut end = start + 1;
-            skip_over_next(slice, &mut end, |ch| matches!(ch, ' ' | '\t'));
+            end = skip_while(slice, end, |ch| matches!(ch, ' ' | '\t')).unwrap_or(end);
```

Comment:
Ah, I think I may have missed this in #135. This should be `is_strict_whitespace` instead of `|ch| matches!(ch, ' ' | '\t'`